docs/en/observability/ci-cd-observability.asciidoc

```diff
@@ -427,3 +427,50 @@ The Concourse CI doesn’t report health metrics through OpenTelemetry. However,
 https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/spanmetricsprocessor#span-metrics-processor[OpenTelemetry Collector Span Metrics Processor]
 to derive pipeline execution traces into KPI metrics like throughput and the error rate
 of pipelines.
+
+
+[discrete]
+[[check-service-health-from-cd-pipelines]]
+== Check service health from deployment pipelines
+
+
+Integrating automated service health checks in deployment pipelines is critical to end-to-end deployment automation which is critical to increase the deployment frequency .
+
+Elastic Observability exposes service HTTP health check APIs that that integrate in deployment pipelines to verify the behaviour of newly deployed instances
+and automatically continue the deployments or rollback according to the health statuses.
+
+
+image::images/ci-cd-canary-deployment-pipeline.png[Canary Deployment Pipeline]
+
+
+The primary HTTP API to check the health of a service is compare the transaction error rate with a threshold value.
+The API call requires authentication, we recomend to use an API Token to authenticate.
+The URL is `KIBANA_URL//internal/apm/services`, it is desing to query some service metrics.
+In order to check the transaction error rate, we have to pass some parameters to the API call.
+We need a rande of time, this range is defined by the parameters `start` and `end`, 
+these parameters are two dates in ISO-8601 format. Decause we want to query only one service,
+we have to compose a filter in the param `kuery`, we can filter by the service name using the expresion `service.name:MY_SERVICE_NAME`.
+Finally we apply a filter by environment passing the parameter `environment`, 
+to select all environments the value is `ENVIRONMENT_ALL`
+
+The API is subject to changes and a stable API optimized for Continous Delivery use cases will soon be published.
+
+[source,python]
+----
+def check_service_health(service_name, error_rate_threshold, kibana_url, api_token):
+    now = datetime.now()
+    five_minutes_ago = now - timedelta(minutes=5)
+    params = {
+        "start": five_minutes_ago.strftime("%Y-%m-%dT%H:%M:%SZ"),
+        "end": now.strftime("%Y-%m-%dT%H:%M:%SZ"),
+        "kuery": "", # "service.name:{}".format(service_name),
```

Comment:
@kuisathaverat don't we need to filter on the `service.version` of the newly deployed canary instances?

Reply:
It picks 5 minutes, we can filter by service versions, not sure if worth it to make complex the example.

Reply:
During a canary deployment, just few instances run with the canary version, most run with the previous version. I think we have to narrow down our query to only measure the health of the canary instances.
With this rationale, either the canary instances are a different `environment` like `production-canary` versus `production` or we just filter on the new `service.version` assuming this attribute has changed.
I tend to prefer the latter.

Reply:
done